lib/trie/trie.go

```diff
@@ -63,9 +64,15 @@ func (t *Trie) Snapshot() (newTrie *Trie) {
 		}
 	}
 
+	var dbCopy db.Database
+	if t.db != nil {
+		dbCopy = t.db.Copy()
+	}
```

Comment:
I don't think we should be copying the whole DB.  This isn't really possible for pebbledb.  I think we need to track the modifications for nodes with hashed values, and delete the old value from the DB if we end up pruning this trie.  Let me know if you wanna sync over this.

Reply:
I solved this using the `deltas` and `pruner` we already have. Now every time we delete a hashed node that points to a value node we are deleting the value node too

Reply:
Can you point me to the code that deletes the values from disk?

Reply:
At the beginning I tought it was related with [this call](https://github.com/ChainSafe/gossamer/blob/53425ddf1db9619c43a7b19d8193ea52ecdd8c94/lib/trie/trie.go#L124 and [this part](https://github.com/ChainSafe/gossamer/blob/53425ddf1db9619c43a7b19d8193ea52ecdd8c94/dot/state/storage.go#L67-L72) in storage.go but I realized that pruner is [doing nothing](https://github.com/ChainSafe/gossamer/blob/53425ddf1db9619c43a7b19d8193ea52ecdd8c94/dot/state/pruner/pruner.go#L44-L47) so I'll be working on this too